src/cloudsc_cuda/CMakeLists.txt

```diff
@@ -191,6 +192,64 @@ if( HAVE_CLOUDSC_C_CUDA )
     )
     ###
 
+    ###### SCC-CUDA-OPT ####
+    ecbuild_add_library(
+        TARGET dwarf-cloudsc-c-cuda-opt-lib
+        INSTALL_HEADERS LISTED
+        SOURCES
+            cloudsc/yoecldp_c.h
+            cloudsc/load_state.h
+            cloudsc/load_state.cu
+            cloudsc/cloudsc_c_opt.h
+            cloudsc/cloudsc_c_opt.cu
+            cloudsc/cloudsc_driver_opt.h
+            cloudsc/cloudsc_driver_opt.cu
+            cloudsc/cloudsc_validate.h
+            cloudsc/cloudsc_validate.cu
+            cloudsc/mycpu.h
+            cloudsc/mycpu.cu
+        PUBLIC_INCLUDES
+            $<INSTALL_INTERFACE:include>
+            $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/cloudsc>
+        PUBLIC_LIBS
+            $<${HAVE_HDF5}:hdf5::hdf5>
+            $<${HAVE_SERIALBOX}:Serialbox::Serialbox_C>
+            $<${HAVE_OMP}:OpenMP::OpenMP_C>
+        DEFINITIONS
+            ${CLOUDSC_DEFINITIONS}
+    )
+
+    target_include_directories(
+      dwarf-cloudsc-c-cuda-opt-lib
+      PUBLIC
+      ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES}
```

Comment:
What precisely do we need from that?
It would be better to link against `CUDA::cudart` (or similar, see https://cmake.org/cmake/help/latest/module/FindCUDAToolkit.html#cuda-toolkit-rt-lib) as a PUBLIC_LIBS target instead